### PR TITLE
Feature/cbolles/thermistor

### PR DESCRIPTION
**Pull request by cbolles**
Super basic implementation of a thermistor that allows the user to provide the conversion function as a parameter. This should allow flexibility for more complex thermistor conversion functions.

Note. This was implemented using fixed point arithmetic to reduce the overhead added by floats. In the future, once the ADC has the functionality to measure in fixed point the voltage (ADC returns voltage in millivolts rather then as a float) then this can be updated accordingly.

Another note, there seemed to be a bit of a strange bug with the ADC that caused the code to halt (not fault, just stop running main) if there is only one channel. This is why the test has two ADC channels while it only needs 1.

### CODE REVIEW

File: samples/thermistor/main.cpp [hunk 1]
```diff
@@ -12,25 +12,40 @@ namespace IO = EVT::core::IO;
 namespace DEV = EVT::core::DEV;
 namespace time = EVT::core::time;
 
+
+/**
+ * Note, this conversion function is just to show off how a conversion
+ * function will loook like. In actuality, this will be determined through
```
Comment (AarC10) [inline samples/thermistor/main.cpp:17]: Just a minor typo here

**Comment by AarC10** [comment]
Implementation looks good to me

**Comment by ddp5730** [comment]
The ADC bug you described should be resolved here: https://github.com/RIT-EVT/EVT-core/pull/42

File: samples/thermistor/main.cpp (hunk 1 above)
Comment (cbolles) [reply samples/thermistor/main.cpp]: Good catch

**Review by ddp5730 — APPROVED** [review]
Looks great.  It's up to you if you want to merge this or wait for #42 first so you can remove that redundant ADC initialization.

File: samples/thermistor/main.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/**
+ * This example prints out the time as read in from the RTC
+ * every second.
```
Comment (mjmagee991) [inline samples/thermistor/main.cpp:3]: This comment should be updated.

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Looks great. I just caught a documentation mistake.

File: samples/thermistor/main.cpp (hunk 2 above)
Comment (cbolles) [reply samples/thermistor/main.cpp]: Good catch

**Review by mjmagee991 — APPROVED** [review]
Looks great

**Pull request merged**